process/block/preprocess/transactions.go

```diff
@@ -996,6 +996,8 @@ func (txs *transactions) createAndProcessMiniBlocksFromMe(
 
 	miniBlocks := txs.getMiniBlockSliceFromMap(mapMiniBlocks)
 
+	txs.gasHandler.SetTotalGasConsumedInSelfShard(totalGasConsumedInSelfShard)
```

Comment:
rollbacks on a block that contained transactions (and thus, non zero gas consumption) to a block that contained no transactions is not treated.

Reply:
they will be replaced by the already-in-place reindexing block mechanism I guess

Reply:
refactored